store/src/java/com/zimbra/cs/service/FileUploadServlet.java

```diff
@@ -188,6 +217,25 @@
         public long getSize()           { return file == null ? 0 : file.getSize(); }
         public BlobInputStream getBlobInputStream()        { return blobInputStream; }
 
+        public String getExtension(FileItem filename) {
+            String fileExtension = null;
+            try {
+                TikaConfig tikaConfig = TikaConfig.getDefaultConfig();
```

Comment:
please see https://github.com/Zimbra/zm-mailbox/commit/99af5db1c47aa25a043c1f152744f6235fd86074 
Make sure we test address all regressions around Tika.